go/cli/internal/cli/install.go

```diff
@@ -38,20 +41,87 @@ func installChart(ctx context.Context, chartName string, namespace string, regis
 		args = append(args, "--set", setValue)
 	}
 
+	// If a profile is provided, write the embedded YAML to a temp file and pass its path
+	var tmpProfilePath string
+	if profile != "" {
+		profileYAML := profiles.GetProfile(profile)
+		if strings.TrimSpace(profileYAML) != "" {
+			tmpFile, err := os.CreateTemp("", "kagent-profile-*.yaml")
+			if err != nil {
+				return "", fmt.Errorf("failed creating temp profile file: %w", err)
+			}
+			if _, err := tmpFile.WriteString(profileYAML); err != nil {
+				_ = tmpFile.Close()
+				_ = os.Remove(tmpFile.Name())
+				return "", fmt.Errorf("failed writing temp profile file: %w", err)
+			}
+			if err := tmpFile.Close(); err != nil {
+				_ = os.Remove(tmpFile.Name())
+				return "", fmt.Errorf("failed closing temp profile file: %w", err)
+			}
+			tmpProfilePath = tmpFile.Name()
+			args = append(args, "-f", tmpProfilePath)
+		}
+	}
+
 	cmd := exec.CommandContext(ctx, "helm", args...)
 	if byt, err := cmd.CombinedOutput(); err != nil {
+		if tmpProfilePath != "" {
+			_ = os.Remove(tmpProfilePath)
+		}
 		return string(byt), err
 	}
+	if tmpProfilePath != "" {
+		_ = os.Remove(tmpProfilePath)
+	}
 	return "", nil
 }
 
-func InstallCmd(ctx context.Context, cfg *config.Config) *PortForward {
+func InstallCmd(ctx context.Context, cfg *config.Config, profile string) *PortForward {
```

Comment:
Why not just put `profile` into `config`, or maybe have an `InstallConfig`. For example see `InvokeConfig`